app.rb

```diff
@@ -25,126 +30,163 @@ def list_all_books
     people << teacher
-    puts "Person(Teacher) created successfully"
+    puts 'Person(Teacher) created successfully'
   end
 
   def create_book
     book_attributes = yoink_book_attributes
     book = Book.new(book_attributes[:title], book_attributes[:author])
     books << book
-    puts "Book created!"
+    puts 'Book created!'
   end
 
   ##### Book renting
   def create_rental
     book = select_book
     person = select_person
-    puts "Date:"
+    puts 'Date:'
     date = gets.chomp
-    @rentals << Rental.new(date, person, book)
-    puts "Rental created successfully!"
+    rental = Rental.new(date, person, book)
+    @rentals << rental
+    puts 'Rental created successfully!'
   end
 
   def select_book
-    puts "Select a book from the following list by number"
+    puts 'Select a book from the following list by number'
     @books.each_with_index do |book, index|
       puts "#{index} Title: #{book.title}, Author: #{book.author}"
     end
-    puts "Book number:"
+    puts 'Book number:'
     book_index = gets.chomp.to_i
     @books[book_index]
   end
 
   def select_person
-    puts "Select a person from the following list by number (not id)"
+    puts 'Select a person from the following list by number (not id)'
     @people.each_with_index do |person, index|
       puts "#{index} [#{person.class}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"
     end
-    puts "Person number:"
+    puts 'Person number:'
     person_index = gets.chomp.to_i
     @people[person_index]
   end
 
   def list_rental_for_person
     id = yoink_id
-    puts "Rentals:"
-    rentals.select { |rental| rental.person.id == id }.each do |rental|
-      puts "Date: #{rental.date}, Book '#{rental.book.title}' by #{rental.book.author}"
+    puts 'Rentals:'
+    @rentals.each do |rental|
+      if rental&.person && rental.person.id.to_i == id.to_i
+        puts "Date: #{rental.date}, Book '#{rental.book.title}' by #{rental.book.author}"
+      else
+        puts 'This person has no book rentals'
+      end
     end
   end
 
+  def save_data
+    File.write('books.json', @books.map(&:to_h).to_json)
+    File.write('rentals.json', @rentals.map(&:to_h).to_json)
+    File.write('people.json', @people.map(&:to_h).to_json)
+  end
+
+  def load_data
+    books_json = File.read('books.json')
+    books_data = JSON.parse(books_json)
+    books_data.each do |book_data|
+      book = Book.new(book_data['title'], book_data['author'])
+      @books << book
+    end
+
+    people_data = JSON.parse(File.read('people.json'))
+    people_data.each do |person_data|
+      person = Person.new(person_data['age'], person_data['name'])
+      @people << person
+    end
+
+    rentals_data = JSON.parse(File.read('rentals.json'))
+    rentals_data.each do |rental|
+      date = rental['date']
+      person = @people.find { |p| p.id == rental['person_id'] }
+      book = @books.find { |b| b.id == rental['book_id'] }
+      @rentals << Rental.new(date, person, book)
+    end
```

Comment:
- [ ] Kindly make sure that the rentals are preserved correctly after the app is reopened. Currently, when I try to save a rental and show it after I reopen the app, I get the following result:

![image](https://user-images.githubusercontent.com/62290545/212200572-b7908f62-1153-4172-82df-4c2062b8907b.png)

Reply:
- It's fixed now rentals is preserved correctly when reopened and you can get the saved person id by listing all people. 😄